src/dfx/src/lib/builders/custom.rs

```diff
@@ -190,12 +192,16 @@ impl CanisterBuilder for CustomBuilder {
     }
 }
 
-fn run_command(args: Vec<String>, vars: &[super::Env<'_>]) -> DfxResult<()> {
+fn run_command(args: Vec<String>, vars: &[super::Env<'_>], cwd: &Path) -> DfxResult<()> {
     let (command_name, arguments) = args.split_first().unwrap();
-
-    let mut cmd = std::process::Command::new(command_name);
+    let canonicalized = which::which_in(command_name, env::var_os("PATH"), cwd)
```

Comment:
If a project's dfx.json has `"build": "build.sh"`, and there is a `build.sh` in that project, and the user has `$HOME/bin` in their path, and `$HOME/bin/build.sh` exists, which `build.sh` should we expect to be called?

Reply:
And, whichever is the answer, let's have tests that demonstrate the behavior.

Reply:
We would expect the local command to be preferred for reproducibility across machines.

Reply:
(I ask because it looks like the code would first search $PATH)